helpers/mysql/test/helpers/mysql_test.rb

```diff
@@ -0,0 +1,147 @@
+
+    describe 'when config[:span_name] is :db_name' do
+      let(:span_name) { :db_name }
+
+      it 'returns database name' do
+        assert_equal(database_span_name, database_name)
+      end
+    end
+
+    describe 'when config[:span_name] is :db_operation_and_name' do
+      let(:span_name) { :db_operation_and_name }
+
+      it 'returns db operation and name' do
+        assert_equal(database_span_name, 'operation database_name')
+      end
+    end
+
+    describe 'when config[:span_name] does not match a case' do
+      let(:span_name) { 'something_unexpected' }
+
+      it 'returns mysql' do
+        assert_equal(database_span_name, 'mysql')
+      end
+    end
+
+    describe 'when config[:span_name] is nil' do
+      let(:span_name) { nil }
+
+      it 'returns mysql' do
+        assert_equal(database_span_name, 'mysql')
+      end
+    end
+  end
+
+  describe '.db_operation_and_name' do
+    let(:operation) { 'operation' }
+    let(:database_name) { 'database_name' }
+    let(:db_operation_and_name) { OpenTelemetry::Helpers::MySQL.db_operation_and_name(operation, database_name) }
+
+    describe 'when operation and database_name are present' do
+      it 'returns a combination of the operation and database_name' do
+        assert_equal(db_operation_and_name, "#{operation} #{database_name}")
+      end
+    end
+
+    describe 'when operation is nil' do
+      let(:operation) { nil }
+
+      it 'returns database_name' do
+        assert_equal(db_operation_and_name, database_name)
+      end
+    end
+
+    describe 'when database_name is nil' do
+      let(:database_name) { nil }
+
+      it 'returns the operation name' do
+        assert_equal(db_operation_and_name, operation)
+      end
+    end
+  end
+
+  describe '.extract_statement_type' do
+    let(:sql) { 'SELECT * FROM users' }
+    let(:extract_statement_type) { OpenTelemetry::Helpers::MySQL.extract_statement_type(sql) }
+
+    describe 'when it finds a match' do
+      it 'returns the query name' do
+        assert_equal('select', extract_statement_type)
+      end
+    end
+
+    describe 'when sql contains invalid byte sequences' do
+      let(:sql) { "SELECT * from users where users.id = 1 and users.email = 'test@test.com\255'" }
+
+      it 'extracts the statement' do
+        assert_equal('select', extract_statement_type)
+      end
+    end
+
+    describe 'when sql contains unknown query statement' do
+      let(:sql) { 'DESELECT 1' }
+
+      # nil sets the span name to 'mysql'
+      it 'returns nil' do
+        assert_nil(extract_statement_type)
+      end
+    end
+
+    describe 'when sql has marginalia-style prepended comments' do
+      let(:sql) do
+        '/*application:web,controller:blob,action:show,correlation_id:01EZVMR923313VV44ZJDJ7PMEZ,' \
+          'endpoint_id:Projects::BlobController#show*/ SELECT "routes".* FROM "routes" WHERE "routes"' \
+          '."source_id" = 75 AND "routes"."source_type" = \'Namespace\' LIMIT 1'
+      end
```

Comment:
I played around with this a bit. While the example above works as expected, the result is incorrect if one of the query names is in the Marginalia comment. E.g.:
```ruby
irb(main):001:1*       QUERY_NAMES = [
irb(main):002:1*         'set names',
irb(main):003:1*         'select',
irb(main):004:1*         'insert',
irb(main):005:1*         'update',
irb(main):006:1*         'delete',
irb(main):007:1*         'begin',
irb(main):008:1*         'commit',
irb(main):009:1*         'rollback',
irb(main):010:1*         'savepoint',
irb(main):011:1*         'release savepoint',
irb(main):012:1*         'explain',
irb(main):013:1*         'drop database',
irb(main):014:1*         'drop table',
irb(main):015:1*         'create database',
irb(main):016:1*         'create table'
irb(main):017:0>       ].freeze
=> 
["set names",
...
irb(main):018:0>       QUERY_NAME_REGEX = Regexp.new("\\b(#{QUERY_NAMES.join('|')})\\b", Regexp::IGNORECASE)
=> /\b(set names|select|insert|update|delete|begin|commit|rollback|savepoint|release savepoint|explain|drop database|drop table|create da...
irb(main):019:0* sql = '/*application:web,controller:blob,action:show,correlation_id:01EZVMR923313VV44ZJDJ7PMEZ,' \
irb(main):020:0* 'endpoint_id:Projects::BlobController#show*/ SELECT "routes".* FROM "routes" WHERE "routes"' \
irb(main):021:0>           '."source_id" = 75 AND "routes"."source_type" = \'Namespace\' LIMIT 1'
=> "/*application:web,controller:blob,action:show,correlation_id:01EZVMR923313VV44ZJDJ7PMEZ,endpoint_id:Projects::BlobController#show*/ S...
irb(main):022:0> QUERY_NAME_REGEX.match(sql) { |match| match[1].downcase } 
=> "select"
irb(main):023:0* sql = '/*application:web,controller:insert,action:show,correlation_id:01EZVMR923313VV44ZJDJ7PMEZ,' \
irb(main):024:0* 'endpoint_id:Projects::BlobController#show*/ SELECT "routes".* FROM "routes" WHERE "routes"' \
irb(main):025:0>           '."source_id" = 75 AND "routes"."source_type" = \'Namespace\' LIMIT 1'
=> "/*application:web,controller:insert,action:show,correlation_id:01EZVMR923313VV44ZJDJ7PMEZ,endpoint_id:Projects::BlobController#show*/...
irb(main):026:0> QUERY_NAME_REGEX.match(sql) { |match| match[1].downcase } 
=> "insert"
irb(main):027:0> 
```

Reply:
Great catch. I've updated the tests to include this case. It's fixed in: 82584bae3c9132846fdd593a124dc680b770af28